From a840b7562668587dd70418aa7279268c41eb8424 Mon Sep 17 00:00:00 2001
From: Federico Mena Quintero <federico@gnome.org>
Date: Thu, 9 May 2024 10:46:33 -0600
Subject: [PATCH] Allow dead code for CairoContextState, since the logging
 machinery doesn't use it yet

Newer rustc complains about this, since that struct is never
constructed outside the tests.

Part-of: <https://gitlab.gnome.org/GNOME/librsvg/-/merge_requests/971>
---
 rsvg/src/log.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rsvg/src/log.rs b/rsvg/src/log.rs
index 15e947bc4..9ad7c7e4f 100644
--- a/rsvg/src/log.rs
+++ b/rsvg/src/log.rs
@@ -20,6 +20,7 @@ macro_rules! rsvg_log {
 /// context as it was passed in by the caller.  For example, librsvg may decide to
 /// operate differently depending on the context's target surface type, or its current
 /// transformation matrix.  This struct captures that sort of information.
+#[allow(dead_code)] // this is never constructed yet; allow building on newer rustc which warns about this
 #[derive(Copy, Clone, Debug, PartialEq)]
 struct CairoContextState {
     surface_type: cairo::SurfaceType,
-- 
GitLab

